src/servers/graphql-main-server.ts

```diff
@@ -56,7 +57,10 @@ if (require.main === module) {
   setupMongoConnection(true)
     .then(async () => {
       activateLndHealthCheck()
-      await startApolloServerForCoreSchema()
+      await Promise.race([
```

Comment:
Why not `Promise.all` here?

Reply:
because I think we want to quit as soon as one critical/non recoverable arise?

Reply:
maybe this should be the behavior for an error, but not for a clean shutdown, in that case we should wait for both servers to stop?

Reply:
Ah right, I hadn't thought about if either of those promises gets rejected. I'm not actually sure what graceful shutdown looks like for these servers currently. Maybe @dolcalmi has some ideas too